numpy/core/src/multiarray/number.c

```diff
@@ -797,6 +797,12 @@ _array_nonzero(PyArrayObject *mp)
         return res;
     }
     else if (n == 0) {
+        if (DEPRECATE("Returning False. "
+                      "However, the truth value of an empty array is ambiguous "
+                      "and will result in an error in the future. "
+                      "Use a.size > 0") < 0) {
```

Comment:
I think this should be reordered to

    The truth value of an empty array is ambiguous.
    Returning `False`, but in future this will result in an error.
    To check for emptiness, use `a.size > 0` instead.

Remember that warnings are not very good at pointing to their origin - so saying what the problem is should come first